tests/service_testing/agent_test_case_test.py

```diff
@@ -91,31 +91,79 @@ def test_assertVerifyResults_failed(self):
         good_results=[], bad_results=[], failed_constraints=[])
     self.assertRaises(AssertionError, self.assertVerifyResults, verify_results)
 
-  def test_run_test_case_ok(self):
+  def _do_run_test_case(self, succeed, with_callbacks, with_context):
+    # pylint: disable=unused-argument
     operation = FakeOperation('TestOperation', self.testing_agent)
 
-    verifier = FakeVerifier(True)
-    clause = jc.ContractClause(
-        'TestClause', observer=FakeObserver(), verifier=verifier)
-    contract = jc.Contract()
-    contract.add_clause(clause)
-
-    operation_contract = st.OperationContract(operation, contract)
-    self.run_test_case(operation_contract)
-
-  def test_run_test_case_failed(self):
-    context = ExecutionContext()
-    operation = FakeOperation('TestOperation', self.testing_agent)
-
-    verifier = FakeVerifier(False)
+    verifier = FakeVerifier(succeed)
     clause = jc.ContractClause(
         'TestClause', observer=FakeObserver(), verifier=verifier)
     contract = jc.Contract()
     contract.add_clause(clause)
 
-    operation_contract = st.OperationContract(operation, contract)
-    self.assertRaises(AssertionError,
-                      self.run_test_case, operation_contract, context=context)
+    class HelperClass(object):
+      """Need to share state between these helper methods and outer scope.
+
+      This class provides the functions we are going to inject, along
+      with state we can check in the outer scope.
+      """
+      cleanup_calls = 0
+      execution_context = ExecutionContext() if with_context else None
+
+      @staticmethod
+      def status_extractor(status, context):
+        if with_context:
+          # Verify this is the context we injected.
+          self.assertEquals(HelperClass.execution_context, context)
+        self.assertIsNotNone(context.get('OperationStatus', None))
+        self.assertIsNone(context.get('GOT_STATUS', None))
+        context['GOT_STATUS'] = status
```

Comment:
It's already there in the status. This is just the interface. Actual status is specialized by the agent that knows how to interpret it. The HttpAgent has a "detail" attribute which is the output of the server, though it . The SpinnakerAgent decodes this into json and provides an attribute "detail_doc". So for tests using SpinnakerAgent (e.g. using the GateAgent), the status of these operations will have a detail_doc with a dictionary form of the JSON response.